views/audit/query_audit_v2.sql

```diff
@@ -0,0 +1,331 @@
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.jobInsertion.job.jobConfig.extractConfig.destinationUris' ) AS destinationUris,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.jobInsertion.job.jobConfig.extractConfig.sourceTable' ) AS sourceTable,
+      SPLIT(JSON_EXTRACT(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.extractConfig.destinationUris'),".")[
+    OFFSET
+      (0)] AS srctable_projectid,
+      SPLIT(JSON_EXTRACT(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.extractConfig.destinationUris'),".")[
+    OFFSET
+      (1)] AS srctable_datasetid,
+      SPLIT(JSON_EXTRACT(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.extractConfig.destinationUris'),".")[
+    OFFSET
+      (2)] AS srctable_tableid,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.jobInsertion.job.jobConfig.queryConfig.query.query' ) AS jobconfig_query,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.jobInsertion.job.jobStatus.errorResult.code' ) AS errorCode,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.jobInsertion.job.jobStatus.errorResult.message' ) AS errorMessage,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        "$.jobChange.after") AS jobChangeAfter,
+      COALESCE(JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.queryConfig.statementType'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.queryConfig.statementType') )AS statementType,
+      REGEXP_EXTRACT(protopayload_auditlog.metadataJson, r'BigQueryAuditMetadata","(.*?)":') AS eventName,
+      
+      COALESCE(JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.labels.querytype'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.labels.querytype')) AS querytype
+          
+         
+    FROM
+      `project_id.dataset_id.table_id` ),
+    BQAudit2_data AS(
+    SELECT
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.tableDataChange.insertedRowsCount') AS insertRowCount,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.tableDataChange.deletedRowsCount') AS deleteRowCount,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+        '$.tableDataChange.reason') AS tableDataChangeReason,
+      CONCAT(SPLIT(JSON_EXTRACT(protopayload_auditlog.metadataJson,
+            '$.tableDataChange.jobName'),"/")[
+      OFFSET
+        (1)], ":", SPLIT(JSON_EXTRACT(protopayload_auditlog.metadataJson,
+            '$.tableDataChange.jobName'),"/")[
+      OFFSET
+        (3)]) AS data_jobid
+    FROM
+      `project_id.dataset_id.table_id`) /* This code queries BQAudit2 */
```

Comment:
This is an implicit CROSS JOIN and is very inefficient. Just add these columns into the previous select